tensorflow-framework/src/main/java/org/tensorflow/framework/optimizers/Optimizer.java

```diff
@@ -56,70 +63,68 @@ protected Optimizer(Graph graph) {
     this.tf = Ops.create(graph).withName(getOptimizerName());
     this.slots = new HashMap<>();
     this.globals = new ArrayList<>();
+    setLearningRate(LEARNING_RATE_DEFAULT);
   }
 
   /**
    * Builds an optimizer for the supplied graph.
    *
+   * <p>Uses the name from {@link Optimizer#getOptimizerName()} to name the operations.
+   *
    * @param graph The graph to optimize.
-   * @param name The base name for the operations.
+   * @param learningRate the learning rate.
    */
-  protected Optimizer(Graph graph, String name) {
+  protected Optimizer(Graph graph, float learningRate) {
```

Comment:
Can we have both of these constructors call into the `Optimizer(Graph,float,String)` one?

Reply:
Being that Optimizer is abstract, we really only need one constructor, `protected Optimizer(Graph graph, String name, float learningRate)` . Of course, we would have to handle a `null` name, with something like:
`this.tf = Ops.create(graph).withName(name == null? getOptimizerName() : name);`

Reply:
CTORS have been changed